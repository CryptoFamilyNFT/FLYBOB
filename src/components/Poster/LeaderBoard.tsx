import React, { useEffect, useState } from 'react';
import { makeStyles, useMediaQuery, useTheme } from '@material-ui/core';
import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Typography,
} from '@mui/material'
import Api from '../../api/api';

const useStyles = makeStyles((theme: any) => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
    overflowX: 'auto',
  },
  table: {
    minWidth: 650,
    fontFamily: "Josefin Sans, sans-serif"
  },
  title: {
    textAlign: 'center',
    marginBottom: theme.spacing(2),
    marginTop: theme.spacing(2),
    fontFamily: "Josefin Sans, sans-serif"
  },
}));

interface Score {
  player: string;
  score: string;
}

const LeaderBoard = () => {
  const classes = useStyles();
  const theme = useTheme()
  const [top10Scores, setTop10Scores] = useState<Score[]>();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

  const getLeaderBoard = async () => {
    try {
      const scores = await Api.getScores();
      const sortedScores = scores.sort((a, b) => parseInt(b.score) - parseInt(a.score));
      const top10 = sortedScores.slice(0, 10);
      setTop10Scores(top10);
    } catch (e) {
      console.log(e);
      setTop10Scores([{ player: '', score: '' }]);
    }
  };

  useEffect(() => {
    getLeaderBoard();
  }, []);

  useEffect(() => {
    const intervalId = setInterval(() => {
      getLeaderBoard();
    }, 20000);

    return () => clearInterval(intervalId);
  }, [top10Scores]);


  return (
    <>
      {!isMobile && (
        <Paper style={{ backgroundColor: 'yellow', minHeight: 300, maxHeight: 300, overflowY: 'auto', gap: 10, border: '1px solid aqua' }}>
          <Typography variant="h4" className={classes.title} style={{ fontFamily: 'Josefin Sans, sans-serif', marginTop: 30 }}>
            Leaderboard
          </Typography>
          <TableContainer>
            <Table className={classes.table} aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>#</TableCell>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>User</TableCell>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>Score</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {top10Scores?.map((row, index) => (
                  <TableRow key={index}>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }} component="th" scope="row">
                      {index + 1}
                    </TableCell>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>{row.player?.slice(0, 5) + '...' + row.player?.slice(-5)}</TableCell>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>{row.score}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        </Paper>
      )}
      {isMobile && (
        <Paper style={{ backgroundColor: 'yellow', maxHeight: 280, overflowY: 'auto', gap: 10, border: '1px solid aqua', width: 360 }}>
          <Typography variant="h4" className={classes.title} style={{ fontFamily: 'Josefin Sans, sans-serif', marginTop: 30 }}>
            Leaderboard
          </Typography>
          <TableContainer>
            <Table className={classes.table} aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>#</TableCell>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>User</TableCell>
                  <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>Score</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {top10Scores?.map((row, index) => (
                  <TableRow key={index}>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }} component="th" scope="row">
                      {index + 1}
                    </TableCell>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>{row.player?.slice(0, 5) + '...' + row.player?.slice(-5)}</TableCell>
                    <TableCell style={{ fontFamily: 'Josefin Sans, sans-serif' }}>{row.score}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        </Paper>
      )}
    </>
  );
};

export default LeaderBoard;
