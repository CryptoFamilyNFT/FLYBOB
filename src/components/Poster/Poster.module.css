
.Poster{
  width: 100%;
  height: 100%;
  border-right: 2px solid aqua;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: aqua;
  background-image: url("../../assets/images/logo.jpeg");
}


.PosterBlur {
  width: 100%;
  height: 100%;
  border-right: 2px solid aqua;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url("../../assets/images/logo.jpeg");
}

.Poster img {
  height: 100%;
}

.PosterMobile {
  height: 50%;
  width: 100;

}

@media screen and (min-width: 768px) {
  .Poster {
    width: 50vw;
    height: 100vh;
  }

  .Poster img {
    width: 100%;
    height: auto;
  }
}
