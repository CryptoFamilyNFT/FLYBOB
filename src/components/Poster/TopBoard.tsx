import React from 'react';
import { makeStyles } from '@material-ui/core';
import {
    Paper,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
} from '@mui/material'

import LOGO from '../../assets/images/drip.jpeg';

const useStyles = makeStyles((theme: any) => ({
    root: {
        width: '100%',
        marginTop: theme.spacing(3),
        overflowX: 'auto',
    },
    table: {
        minWidth: 650,
        fontFamily: "Josefin Sans, sans-serif"
    },
    title: {
        textAlign: 'center',
        marginBottom: theme.spacing(2),
        fontFamily: "Josefin Sans, sans-serif"
    },
}));

const data = [
    { user: 'User1', score: 100 },
    { user: 'User2', score: 90 },
    { user: 'User3', score: 100 },
    { user: 'User4', score: 90 },
];

const TopBoard = () => {
    const classes = useStyles();
    return (
        <div>
            <div style={{ backgroundColor: 'transparent', marginTop: 40, border: '2px solid transparent', borderBottom: 'none'}}>
                <img src={LOGO} alt="" style={{ border: '2px solid transparent', borderRadius: '50%', width: 200, height: 200 }} />
                <Typography style={{ color: 'yellow', fontFamily: "Josefin Sans, sand-serif", fontWeight: 'bold', fontSize: '40px', marginTop: 50, textShadow: '0 0 20px black' }}> BOB ADVENTURES 2</Typography>
            </div>
        </div>
    );
};

export default TopBoard;
