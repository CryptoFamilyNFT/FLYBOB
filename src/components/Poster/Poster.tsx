// Libs
import React, { Suspense } from 'react';

// Images
import Logo from '../../assets/images/logo.jpeg';

// Styles
import styles from './Poster.module.css';
import { Link, useLocation } from 'react-router-dom';

// Constants
import testAttributes from "../../testAttributes";
import Leaderboard from './LeaderBoard';
import { Button, useMediaQuery, useTheme } from '@material-ui/core';
import { Route, Switch, withRouter } from 'react-router-dom';
import TopBoard from './TopBoard';

const leaderboardData = [
  { user: 'User1', score: 100 },
  { user: 'User2', score: 90 },
  { user: 'User3', score: 100 },
  { user: 'User4', score: 90 },
];

const Poster = () => {
  const theme = useTheme()
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  const routes = (<Switch>
    <Route path='/leaderboard' component={Leaderboard} />
    <Route path='/topboard' component={TopBoard} />
  </Switch>);
  const location = useLocation();

  const fallbackNode = <p>Loading...</p>;

  return (
    <div className={styles.Poster} data-testid={testAttributes.POSTER_COMPONENT}>
      {isMobile && location.pathname !== '/' ? (
        <div className={styles.PosterMobile}>
          <Suspense fallback={fallbackNode}>{routes}</Suspense>
        </div>
      ) : (
        <div>
          {isMobile && location.pathname === '/' ? <TopBoard/> : <Leaderboard />}
        </div>
      )}
    </div>
  );
};

export default Poster;
