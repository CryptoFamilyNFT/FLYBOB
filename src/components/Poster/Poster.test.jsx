// Libs
import React from 'react';
import renderer from 'react-test-renderer';

// Components
import Poster from './Poster';

describe('<Poster />', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(<Poster/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

