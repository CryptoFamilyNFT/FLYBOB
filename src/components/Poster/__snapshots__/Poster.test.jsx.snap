// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poster /> should renders correctly 1`] = `
<div
  className="Poster"
  data-testid="Poster"
>
  <img
    alt="Logo"
    loading="lazy"
    src="logo.png"
  />
</div>
`;
